### PR TITLE
Add mock tables with specs

### PR DESCRIPTION
**Pull request by alissapajer**

### CODE REVIEW

**Comment by alissapajer** [comment]
This doesn't include tests for preparation because, without having more of the actual preparation implementation complete, I wasn't sure exactly how to mock out the various preparation states and still have a meaningful test. I'll come back to this when the actual preparation implementation is added.

**Comment by alissapajer** [comment]
[ch452]

File: api/src/main/scala/quasar/api/table/Table.scala [hunk 1]
```diff
@@ -16,8 +16,31 @@
...
+object TableName {
+  implicit val equalTableName: Equal[TableName] = Equal.equalA
+  implicit val showTableName: Show[TableName] = Show.showFromToString
+}
+
+object Table {
+  import TableName._
+
+  implicit def equalTable[Q: Equal]: Equal[Table[Q]] =
+    Equal.equal {
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/table/Table.scala]: Alternatively, `Equal.equalBy(t => (t.name, t.query))`

File: api/src/main/scala/quasar/api/table/Table.scala [hunk 2]
```diff
@@ -16,8 +16,31 @@
 
 package quasar.api.table
 
-import slamdata.Predef.String
+import scalaz.{Cord, Equal, Show}
+import scalaz.syntax.equal._
+import scalaz.syntax.show._
+import slamdata.Predef._
```
Comment (wemrysi) [inline api/src/main/scala/quasar/api/table/Table.scala]: Minor, but we usually make `slamdata.Predef` the first import.

File: api/src/test/scala/quasar/api/table/MockTables.scala [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+import quasar.Condition
+import quasar.contrib.scalaz.MonadState_
+import quasar.contrib.std.uuid._
+import scalaz.{\/, IMap, Monad}
+import scalaz.syntax.either._
+import scalaz.syntax.monad._
+import scalaz.syntax.std.option._
+import slamdata.Predef.{List, None, Option, Some, String, Unit}
+
+final class MockTables[F[_]: Monad: MockTables.TablesMockState[?[_]]]
```
Comment (wemrysi) [inline api/src/test/scala/quasar/api/table/MockTables.scala]: Minor, no need for the `?[_]` when the type-constructor is the only type argument.

File: foundation/src/main/scala/quasar/contrib/std/uuid.scala [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import scalaz.{Order, Ordering, Show}
+import scalaz.std.anyVal._
+import scalaz.syntax.equal._
+
+object uuid {
+
+  implicit val uuidOrder: Order[UUID] = new Order[UUID] {
+    def order(x: UUID, y: UUID): Ordering = {
+      val compare = x.compareTo(y)
```
Comment (wemrysi) [inline foundation/src/main/scala/quasar/contrib/std/uuid.scala]: Alternatively, `Ordering.fromInt(x.compareTo(y))`.

File: api/src/test/scala/quasar/api/table/TablesSpec.scala [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+abstract class TablesSpec[F[_]: Monad: Sync, G[_], I: Equal: Show, Q: Equal: Show, D]
+    extends Qspec
+    with ConditionMatchers {
+
+  import Table._
+
+  def tables: Tables[F, G, I, Q, D]
+
+  // `table1` and `table2` must have distinct names
```
Comment (wemrysi) [inline api/src/test/scala/quasar/api/table/TablesSpec.scala]: Not necessary, but you could add an example that verifies this so the test will fail if they ever happen to be the same.

Comment (alissapajer) [reply api/src/test/scala/quasar/api/table/TablesSpec.scala]: Good idea.

File: api/src/main/scala/quasar/api/table/Table.scala (hunk 2 above)
Comment (alissapajer) [reply api/src/main/scala/quasar/api/table/Table.scala]: All the imports in `quasar.api.table` were strictly alphabetized. I fixed them to comply with what we normally do in quasar.

**Comment by alissapajer** [comment]
@wemrysi ready for another look (pending CI).

File: foundation/src/test/scala/quasar/QuasarSpecification.scala [hunk 6]
```diff
@@ -55,6 +55,7 @@ trait QuasarSpecification extends AnyRef
 
   implicit class Specs2ScalazOps[A : Equal : Show](lhs: A) {
     def must_=(rhs: A) = lhs must equal(rhs)
+    def must_!=(rhs: A) = lhs must not(equal(rhs))
```
Comment (wemrysi) [inline foundation/src/test/scala/quasar/QuasarSpecification.scala]: :+1:

**Pull request merged**